tendermint/src/lite/verifier.rs

```diff
@@ -29,11 +29,15 @@ where
     match now.duration_since(last_header.bft_time().into()) {
         Ok(passed) => {
             if passed > *trusting_period {
-                return Err(Error::Expired);
+                return Err(Kind::Expired {
+                    at: last_header.bft_time().into() + *trusting_period,
```

Comment:
Would it be clearer to have some variable like `expires_at = ...` for this and then check `if expires_at < now` instead of using `passed` ?

Reply:
Will address in a followup PR 👍

Reply:
We can do this but then we must additionally check for header_time > now. Good side effect we can simplify  DurationOutOfRange(#[from] SystemTimeError) to simply `DurationOutOfRange` (maybe the name should actually be `HeaderInTheFuture` or sth like this.

Also, shouldn't this be `expires_at <= now` instead of `<`? On equality we have just expired...

Reply:
See: 1833e77